### DISCUSSION TITLE
Using List in Routing

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/anvil-extras#433] [Q&A] [answered]
I have a list of strings I am trying to to pass using the ```url_pattern```,```url_dict``` arguments of the ```routing.set_url_hash``` but when I access the ```url_dict``` in the form routed to, the list is now a string " [ item 1 , item 2  ]".

Is there a simple way to resolve this?

**Comment by lewysigns** [comment]
I had to call ```eval``` on the list in my url_dict to convert them from the string to list

**Reply by stenci** [reply]
Careful! 
Using `eval` on a string that anyone can feed to your app is very dangerous. I could go to this url: `yourapp.com/#list=return password` and see your password (assuming that there is a variable called `password`). 

It's safe to write your own function to parse the url arguments.

**Reply by lewysigns** [reply, marked as answer]
Thanks.

I wrote this instead, leveraging what I know about the list:

```python
def convert_string_list(string_list):
  try:
    # Remove surrounding brackets and whitespace
    string_list = string_list.strip()[1:-1]

    # Split the string by commas
    elements = string_list.split(',')

    # Remove leading/trailing whitespace from each element
    elements = [element.strip().strip("'") for element in elements]

    # Return the list of strings
    return elements
  except Exception as e:
    print(f"Error: {e}")
    return None
```

**Reply by lewysigns** [reply]
Found a better solution. 

I originally tried to use json.loads, but it was failing because the url_dict uses single quotes. 

I updated my method to do the following:

```python
import jsondef convert_string(string_list):
  string = string_list.replace("'", "\"")
  if string == "None":
    return None
  return json.loads(string)
```
